ApplicationService/repositories/connectionInterface.py

```diff
@@ -1,12 +1,15 @@
-from typing import Protocol
```

Comment:
Why connection_pool and cursor are on ApplicationService/repositories? Both are implementation detail of actual repositories and db context, but are not essential to use case